server/index.js

```diff
@@ -57,6 +60,106 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json([]);
+  } else {
+    const cartId = req.session.cartId;
+    const sql = `
+      select "ci"."cartItemId",
+             "ci"."price",
+             "p"."productId",
+             "p"."image",
+             "p"."name",
+             "p"."shortDescription"
+        from "cartItems" as "ci"
+        join "products" as "p" using ("productId")
+       where "ci"."cartId" = $1
+      `;
+    const params = [cartId];
+
+    db.query(sql, params)
+      .then(result => {
+        res.status(200).json(result.rows[0]);
+      })
+      .catch(err => console.error(err));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+  if (typeof productId === 'undefined') {
+    res.status(400).json({
+      error: 'Missing required value'
+    });
+  } else if (productId < 0 || isNaN(productId)) {
+    res.status(400).json({
+      error: `${productId} must be valid integer`
+    });
+  }
+
+  const sql = `
+    select "price"
+    from "products"
+    where "productId" = $1;
+  `;
+  const value = [productId];
+
+  db.query(sql, value)
+    .then(result => {
+      if (result.rows.length === 0) {
+        throw (new ClientError(`ProductId ${productId} does not exist`, 400));
+      }
+
+      const price = result.rows[0].price;
+      const insertSql = `
+        insert into "carts" ("cartId", "createdAt")
+        values (default, default)
+        returning "cartId"
+      `;
+
+      return db.query(insertSql)
+        .then(insertResult => {
+          const cartId = insertResult.rows[0].cartId;
+          return ({
+            cartId: cartId,
+            price: price
+          });
+        })
+        .catch(err => console.error(err));
+    })
+    .then(result => {
+      req.session.cartId = result.cartId;
+      const sql = `
+        insert into "cartItems" ("cartId", "productId", "price")
+        values ($1, $2, $3)
+        returning "cartItemId"
+      `;
+      const params = [result.cartId, productId, result.price];
+      return db.query(sql, params);
+    })
+    .then(result => {
+      const cartItemId = result.rows[0].cartItemId;
+      const sql = `
+        select "ci"."cartItemId",
+               "ci"."price",
+               "p"."productId",
+               "p"."image",
+               "p"."name",
+               "p"."shortDescription"
+          from "cartItems" as "ci"
+          join "products" as "p" using ("productId")
+         where "ci"."cartItemId" = $1
+      `;
+      const params = [cartItemId];
+      db.query(sql, params)
```

Comment:
Be sure to `return` this `db.query()` so that if the query fails, your final `.catch()` will handle it.